models/darknet/darknet.hpp

```diff
@@ -47,7 +47,8 @@
 #include <mlpack/methods/ann/init_rules/glorot_init.hpp>
 
 namespace mlpack {
-namespace ann /** Artificial Neural Network. */{
+namespace ann /* Artificial neural networks */{
+namespace models {
```

Comment:
Personally I think it might be cleaner to put the various networks in `mlpack::models`, instead of `mlpack::ann::models`.  Let me know what you think. :+1:

Reply:
So just having mlpack::models was creating a issue that i cannot just use the ann namespace inside the namespace by doing the usual `using namespace mlpack::ann` and if i can't do that i have to go everywhere and use the the whole naming convention like `mlpack::ann::component_i_want_to_use` so to avoid huge code changes I used the ann namespace too. 
Is there a workaround?

Reply:
I think that just `ann::component_i_want_to_use` should work too, since both are under the `mlpack` namespace.  But still, I see your point.  The main reason I suggested `mlpack::models` instead of `mlpack::ann::models` is that someday we might add models to this repository that are not ANN models, and so the `mlpack::ann::models` namespace might not make sense.  (Of course, in that case, we could just use `mlpack::models` there, perhaps?)

I can see advantages and disadvantages to both approaches, but I'm not sure which is better.  From a user perspective, `mlpack::models` is probably easiest and clearest, but from a developer perspective, the disadvantage is that we may need `ann::` in the code.

What do you think?  Also, @kartikdutt18, @zoq, and others, any opinions or thoughts?  I'm pretty on the fence on this one; no strong opinion.

Reply:
I am on the side of `mlpack::model` because of these points 
1. There could be more models that are not ANN
2. Easy for user  

Let me know what others think and i can make the changes.

Reply:
Agreed, the plan was always to use this for everything mlpack implements, the ann part is only a small part, so I think it makes sense if the namespace reflects that.

Reply:
i'll start making the changes.